examples/polypharmacy_gripnet/utils.py

```diff
@@ -1,93 +1,84 @@
-    y = target_tensor.detach().cpu().numpy()
-    pred = score_tensor.detach().cpu().numpy()
-    micro, macro = metrics.f1_score(y, pred, average="micro"), metrics.f1_score(y, pred, average="macro")
+    return SuperVertexParaSetting(sv_configs.NAME, sv_configs.INTER_FEAT_CHANNELS, sv_configs.INTER_AGG_CHANNELS_LIST,)
 
-    return micro, macro
 
+class PolypharmacyDataset(Dataset):
+    """Polypharmacy side effect prediction dataset. Only for full-batch training."""
 
-def acc(target_tensor, score_tensor):
-    y = target_tensor.detach().cpu().numpy()
-    pred = score_tensor.detach().cpu().numpy()
-    return accuracy_score(y, pred)
+    def __init__(self, data: Data, mode: str = "train"):
+        super(PolypharmacyDataset, self).__init__()
+
+        self.edge_index = data.__getitem__(f"{mode}_idx")
+        self.edge_type = data.__getitem__(f"{mode}_et")
+        self.edge_type_range = data.__getitem__(f"{mode}_range")
+
+        self.len = self.edge_type_range.shape[0]
 
+    def __len__(self):
 
-def load_graph(pt_file_path="./sample_graph.pt"):
-    """
-    Parameters
-    ----------
-    pt_file_path : file path
+        return 1
 
-    Returns
-    -------
-    graph : torch_geometric.data.Data
-        - data.n_node: number of nodes
-        - data.n_node_type: number of node types == (1 or 2)
-        - data.n_edge: number of edges
-        - data.n_edge_type: number of edge types
-        - data.node_type: (source_node_type, target_node_type)
+    def __getitem__(self, idx):
 
-        - data.edge_index: [list of] torch.Tensor, int, shape (2, n_edge), [indexed by edge type]
-            [0, :] : source node index
-            [1, :] : target node index
-        - data.edge_type: None or list of torch.Tensor, int, shape (n_edge,), indexed by edge type
-        - data.edge_weight: None or list of torch.Tensor, float, shape (n_edge,)
+        return self.edge_index, self.edge_type, self.edge_type_range
 
-        - data.source_node_idx_to_id: dict {idx : id}
-        - data.target_node_idx_to_id: dict {idx : id}
-    """
 
-    return torch.load(pt_file_path)
+def get_all_dataloader(data: Data):
+    """Get train and test dataloader"""
 
+    dataloader_list = []
+    for mode in ["train", "test"]:
+        dataset = PolypharmacyDataset(data, mode=mode)
+        loader = DataLoader(dataset, batch_size=1)
+        dataloader_list.append(loader)
 
-def load_node_idx_to_id_dict(pkl_file_path="./data/pose-1/map.pkl"):
-    """
-    Parameters:
-    -----------
-    The path of index maps in the dataset directory
+    return dataloader_list
 
-    Returns:
-    --------
-    a dictionary of map from node index to entity id/name
-    """
-    with open(pkl_file_path, "rb") as f:
-        out = pickle.load(f)
-    return out
+
+# ----------------------------------------------------------------------------
+# Copy-paste from https://github.com/NYXFLOWER/GripNet
 
 
 def negative_sampling(pos_edge_index, num_nodes):
```

Comment:
This should be reusable and should go into the core API. Is it specific to Polypharmacy? Can it be used for other graph problems? If the latter, consider move it outside of polypharmacy class.